src/Evolution/Systems/Ccz4/TimeDerivative.cpp

```diff
@@ -0,0 +1,604 @@
+                                  *conformal_christoffel_second_kind);
+
+  // eq 18 - 20
+  ::Ccz4::spatial_ricci_tensor(
+      spatial_ricci_tensor, spatial_ricci_tensor_buffer,
+      *christoffel_second_kind, *d_conformal_christoffel_second_kind,
+      conformal_spatial_metric, *inv_conformal_spatial_metric, field_d,
+      *field_d_up, field_p, d_field_p);
+
+  // eq 21
+  ::Ccz4::grad_grad_lapse(grad_grad_lapse, *lapse, *christoffel_second_kind,
+                          field_a, d_field_a);
+
+  // eq 22
+  ::Ccz4::divergence_lapse(divergence_lapse, *conformal_factor_squared,
+                           *inv_conformal_spatial_metric, *grad_grad_lapse);
+
+  // eq 23
+  ::Ccz4::contracted_conformal_christoffel_second_kind(
+      contracted_conformal_christoffel_second_kind,
+      *inv_conformal_spatial_metric, *conformal_christoffel_second_kind);
+
+  // eq 24
+  ::Ccz4::deriv_contracted_conformal_christoffel_second_kind(
+      d_contracted_conformal_christoffel_second_kind,
+      *inv_conformal_spatial_metric, *field_d_up,
+      *conformal_christoffel_second_kind, *d_conformal_christoffel_second_kind);
+
+  // temp for eq 25
+  ::TensorExpressions::evaluate<ti_I>(
+      gamma_hat_minus_contracted_conformal_christoffel,
+      gamma_hat(ti_I) - (*contracted_conformal_christoffel_second_kind)(ti_I));
+
+  // eq 25
+  ::Ccz4::spatial_z4_constraint(
+      spatial_z4_constraint, conformal_spatial_metric,
+      *gamma_hat_minus_contracted_conformal_christoffel);
+
+  // eq 25
+  ::Ccz4::upper_spatial_z4_constraint(
+      upper_spatial_z4_constraint, upper_spatial_z4_constraint_buffer,
+      *conformal_factor_squared,
+      *gamma_hat_minus_contracted_conformal_christoffel);
+
+  // temp for eq 26
+  ::TensorExpressions::evaluate<ti_i, ti_L>(
+      d_gamma_hat_minus_contracted_conformal_christoffel,
+      d_gamma_hat(ti_i, ti_L) -
+          (*d_contracted_conformal_christoffel_second_kind)(ti_i, ti_L));
+
+  // eq 26
+  ::Ccz4::grad_spatial_z4_constraint(
+      grad_spatial_z4_constraint, *spatial_z4_constraint,
+      conformal_spatial_metric, *christoffel_second_kind, field_d,
+      *gamma_hat_minus_contracted_conformal_christoffel,
+      *d_gamma_hat_minus_contracted_conformal_christoffel);
+
+  // eq 27
+  ::Ccz4::ricci_scalar_plus_divergence_z4_constraint(
+      ricci_scalar_plus_divergence_z4_constraint, *conformal_factor_squared,
+      *inv_conformal_spatial_metric, *spatial_ricci_tensor,
+      *grad_spatial_z4_constraint);
+
+  // temporary expressions not already computed above
+
+  ::TensorExpressions::evaluate(contracted_field_b, field_b(ti_k, ti_K));
+
+  ::TensorExpressions::evaluate<ti_k, ti_j, ti_I>(
+      symmetrized_d_field_b,
+      0.5 * (d_field_b(ti_k, ti_j, ti_I) + d_field_b(ti_j, ti_k, ti_I)));
+
+  ::TensorExpressions::evaluate<ti_k>(
+      contracted_symmetrized_d_field_b,
+      0.5 * (d_field_b(ti_k, ti_l, ti_L) + d_field_b(ti_l, ti_k, ti_L)));
```

Comment:
Thanks for this suggestion! I hadn't thought about computing it this way, and t is definitely less operations to take the trace. For now, I'll rewrite it with `for` loops, because Kronecker delta isn't yet supported (on the TODO list)